File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Bookshelf.kt
package com.storysphere.storyreader.model

data class Bookshelf(
    val id: String,
    val userId: String,
    val name: String,
    val description: String? = null,
    val color: String? = null,
    val icon: String? = null,
    val storyIds: List<String> = emptyList(),
    val order: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val syncStatus: SyncStatus = SyncStatus.SYNCED,
    val lastSyncedAt: Long? = null
)

File: Novels/packages/5-mobile-ios/StoryReader/Model/Bookshelf.swift
import Foundation

// Bookshelf model - Enhanced with items support
struct Bookshelf: Codable, Identifiable {
    let id: String
    let userId: String
    let name: String
    let description: String?
    let color: String?
    let icon: String?
    let displayOrder: Int
    let isDefault: Bool
    let items: [BookshelfItem]?
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case description
        case color
        case icon
        case displayOrder = "display_order"
        case isDefault = "is_default"
        case items
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

// BookshelfItem model
struct BookshelfItem: Codable, Identifiable {
    let id: String
    let bookshelfId: String
    let libraryId: String
    let library: LibraryItem?
    let displayOrder: Int
    let addedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case bookshelfId = "bookshelf_id"
        case libraryId = "library_id"
        case library
        case displayOrder = "display_order"
        case addedAt = "added_at"
    }
}

